types/babel-types/babel-types-tests.ts

```diff
@@ -52,3 +52,64 @@ traverse(ast, {
         }
     }
 });
+
+// TypeScript Types
+// TODO: Test all variants of these functions' signatures
+
+const id = t.identifier("id");
+const tparam = t.typeParameterDeclaration([id]);
+
+const any = t.tSAnyKeyword();
+t.tSArrayType(any);
+t.tSAsExpression(exp, any);
+t.tSBooleanKeyword();
+t.tSCallSignatureDeclaration(tparam);
+t.tSConstructSignatureDeclaration();
+t.tSConstructorType();
+t.tSDeclareFunction(id, t.noop(), [id], t.noop());
+t.tSDeclareMethod([t.decorator(exp)], exp, t.noop(), [id]);
+t.tSEnumDeclaration(id, [t.tSEnumMember(id)]);
+t.tSEnumMember(id);
+const expo = t.tSExportAssignment(exp);
+t.tSExpressionWithTypeArguments(id);
+const ext = t.tSExternalModuleReference(t.stringLiteral());
+t.tSFunctionType();
+t.tSImportEqualsDeclaration(id, ext);
+const sig = t.tSIndexSignature([id]);
+t.tSIndexedAccessType(any, any);
+t.tSInterfaceBody([sig]);
+t.tSInterfaceDeclaration(id, null, null, t.tSInterfaceBody([sig]));
+t.tSIntersectionType([any]);
+t.tSLiteralType(t.stringLiteral("a"));
+t.tSMappedType(t.typeParameter());
+t.tSMethodSignature(id);
+const block = t.tSModuleBlock([expo]);
+t.tSModuleDeclaration(id, block);
+t.tSNamespaceExportDeclaration(id);
+t.tSNeverKeyword();
+t.tSNonNullExpression(exp);
+t.tSNullKeyword();
+t.tSNumberKeyword();
+t.tSObjectKeyword();
+t.tSParameterProperty(id);
+t.tSParenthesizedType(any);
+t.tSPropertySignature(id);
+t.tSQualifiedName(id, id);
+t.tSStringKeyword();
+t.tSSymbolKeyword();
```

Comment:
Do you know where this function is being generated? It would be nice to change the name to `tsSymbolKeyword` instead.

Reply:
See the doc link above - you can definitely file an issue with the Babel team. This API is new so they might be open to the idea!

Reply:
babel/babel#6993